test/SourceKit/InterfaceGen/gen_public_module_name.swift

```diff
@@ -0,0 +1,59 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t --leading-lines
+
+// RUN: %target-swift-frontend %t/common.swift \
+// RUN:   -module-name Common \
+// RUN:   -emit-module-path %t/Common.swiftmodule \
+// RUN:   -emit-module-interface-path %t/Common.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6
+
+// RUN: %target-swift-frontend %t/modimpl.swift -I %t \
+// RUN:   -module-name BestModImpl \
+// RUN:   -emit-module-path %t/BestModImpl.swiftmodule \
+// RUN:   -emit-module-interface-path %t/BestModImpl.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6 \
+// RUN:   -public-module-name BestMod
+
+// RUN: %target-swift-frontend %t/mod.swift -I %t \
+// RUN:   -module-name BestMod \
+// RUN:   -emit-module-path %t/BestMod.swiftmodule \
+// RUN:   -emit-module-interface-path %t/BestMod.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6
+
+// RUN: %sourcekitd-test -req=interface-gen -module BestMod -- -swift-version 6 -I %t -target %target-triple &> %t/BestMod.generatedinterface
+// RUN: %FileCheck -input-file=%t/BestMod.generatedinterface -check-prefix INTERFACE %s
+
+//--- common.swift
+public struct CommonType {}
+
+//--- modimpl.swift
+import struct Common.CommonType
+
+public func fromModImpl() {}
+
+//--- mod.swift
+@_exported import BestModImpl
+import struct Common.CommonType
+
+public func fromMod() {}
+
+// INTERFACE-NOT: import BestModImpl
+// INTERFACE: import struct Common.CommonType
+// INTERFACE-NOT: import
+// INTERFACE: func fromMod()
+// INTERFACE: func fromModImpl()
+
+//--- use.swift
+import BestMod
+
+func test() {
+  // RUN: %sourcekitd-test -req=cursor -pos=%(line+1):3 %t/use.swift -- -swift-version 6 -I %t -target %target-triple %t/use.swift | %FileCheck -check-prefix=MOD %s
+  fromMod()
+  // MOD: fromMod
+  // MOD: BestMod
+
+  // RUN: %sourcekitd-test -req=cursor -pos=%(line+1):3 %t/use.swift -- -swift-version 6 -I %t -target %target-triple %t/use.swift | %FileCheck -check-prefix=MODIMPL %s
+  fromModImpl()
+  // MODIMPL: fromModImpl
+  // MODIMPL: BestMod
```

Comment:
I assume this is checking the module is declared as `BestMod` but not `BestModImpl`. But I think this also matches `BestModImpl`, no? Maybe rename `BestMod` to `BestModPub` or such?

Reply:
How embarrassing. Renamed (and it was indeed failing)